src/OrchardCore.Modules/OrchardCore.Media/Startup.cs

```diff
@@ -391,6 +391,30 @@ public override void ConfigureServices(IServiceCollection services)
 </table>";
                 d.Categories = new string[] { "HTML Content", "Media" };
             });
+            services.AddShortcode<ShareShortCodeProvider>("share", d =>
+            {
+                d.DefaultValue = "[share] [/share]";
+                d.Hint = "Add a link to sharable, downloadable content from the media library.";
+                d.Usage =
+@"[share]foo.jpg[/share]<br>
+<table>
+  <tr>
+    <td>Args:</td>
+    <td>url, save</td>
+  </tr>
+  <tr>
+    <td></td>
+    <td>class, tooltip</td>
+  </tr>
+</table>";
+                d.Categories = new string[] { "HTML Content", "Media" };
+            });
+
+            // whitelist the 'download' attribute to enable the share-shortcode
+            services.ConfigureHtmlSanitizer((sanitizer) =>
```

Comment:
Why not just add "download" as a valid attribute to the sanitizer options? Why is this new class necessary?

Reply:
The class is not entirely necessary.  I spent a little time a few days ago with the liquid templating, and I can get close for the most part to what I might otherwise want, but a certain behavior isn't working like I would expect.  I haven't had a chance to write formal tests to demonstrate it yet, but this should allow you to self-demonstrate.

The content of the shortcode template I used consisted of the following:
```
{% capture output %}
    {% assign fixed_location = Args.url | default: Content | asset_url %}
    <!-- {{ fixed_location }} -->
    <!-- {{ Content}} -->
    {% assign _content = Content | default: fixed_location %}

    {% assign fixed_content = _content | strip | default: fixed_location %}
    <!-- {{ fixed_content }} -->
  	{% assign default_save = Args.save | fixed_content | split: "/" | last %}
    {% assign default_title = Args.tooltip | default: default_save | split: "." | first | capitalize %}
	<a class="{{ Args.classes }}" title="{{ default_title }}" href="{{ fixed_location }}" download="{{ default_save }}">{{ fixed_content }}</a>
{% endcapture %}
{{ output | sanitize | raw }}

```
The HTML content I tested against included this subset of test links:
```
[link url="docs/design.docx" classes="nav-link" tooltip="Design"]Design[/link]<br>
[link url='docs/testing.docx' classes="nav-link"][/link]<br>
[link]https://localhost:44300/TheBlogTheme/img/post-bg.jpg[/link]
[link]rock_and_roll.reunion[/link]
```
This works, except when the `Content` property is null or empty or childless (second item in the above list) and you want to infer the content (and other secondary attributes) from the given `url`.  The liquid filter I would expect to catch a null or empty `Content` does not., so the `default` filter is never exercised.

Reply:
If you think shortcodes are not allowing the scenario you mention in option 2, can you file an issue?

I definitely think we can use a custom shortcode by code. And acutally a `link` and a `url`. 

```
[link url="docs/design.docx" classes="nav-link" tooltip="Design"]Design[/link]
[link url='docs/testing.docx' classes="nav-link" /] // infer text from url
[url]docs/testing.docx[/url] // just the url, resolved for the current tenant
```

And option for the link could be to use the same attributes as <a> (href, ...) and only process `href`. But I kind of like being as natural as possible for shortcodes without thinking about html.